artifact/v3/header/header.cpp

```diff
@@ -87,6 +87,18 @@ ExpectedHeader Parse(io::Reader &reader, ParserConfig conf) {
 
 	tok = lexer.Next();
 	vector<ArtifactScript> state_scripts {};
+	if (tok.type == token::Type::ArtifactScripts) {
+		// Make sure the `scripts` dir exists
+		if (not path::FileExists(conf.artifact_scripts_filesystem_path)) {
+			log::Trace(
+				"Creating the Artifact script directory: " + conf.artifact_scripts_filesystem_path);
+			error::Error err = path::CreateDirectory(conf.artifact_scripts_filesystem_path);
```

Comment:
What about parent directories? `filesystem::create_directory` will not create them for us and hence this fix is incomplete.

Reply:
> It is create directory singular

Yes, I think the helper is good

> I don't think we can run into this situation

The path can be overridden, so we can run into this situation.

I think we either log and error or fix all the way. As it is now in the scenario of a custom path it will error _creating_ the directory rather than running the scripts, which I think is more confusing than helping.

Reply:
If I'm tie breaker, I think I will vote for creating the whole directory chain. This is a function we are likely to reuse as well, and I think this behavior is in general more useful. And it is certainly not wrong to do it in this specific case either.

Reply:
Alright, `mkdir --parents cpp version` coming up :rocket: